skyportal/handlers/api/gwdetector.py

```diff
@@ -0,0 +1,234 @@
+        ---
+        description: Create gwdetector
+        tags:
+          - gwdetectors
+        requestBody:
+          content:
+            application/json:
+              schema: GWDetectorNoID
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New gwdetector ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        data = self.get_json()
+
+        with self.Session() as session:
+
+            schema = GWDetector.__schema__()
+            if 'lat' not in data or 'lon' not in data:
+                return self.error('Missing latitude or longitude')
+            elif not isinstance(data['lat'], (int, float)) or not isinstance(
+                data['lon'], (int, float)
+            ):
+                return self.error('Latitude and longitude must all be numbers')
+            elif (
+                data['lat'] < -90
+                or data['lat'] > 90
+                or data['lon'] < -180
+                or data['lon'] > 180
+            ):
+                return self.error(
+                    'Latitude must be between -90 and 90, longitude between -180 and 180'
+                )
+            try:
+                gwdetector = schema.load(data)
+            except ValidationError as e:
+                return self.error(
+                    'Invalid/missing parameters: ' f'{e.normalized_messages()}'
+                )
+            session.add(gwdetector)
+            session.commit()
+
+            self.push_all(action="skyportal/REFRESH_GWDETECTORS")
+            return self.success(data={"id": gwdetector.id})
+
+    @auth_or_token
+    def get(self, gwdetector_id=None):
+        """
+        ---
+        single:
+          description: Retrieve a gwdetector
+          tags:
+            - gwdetectors
+          parameters:
+            - in: path
+              name: gwdetector_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleGWDetector
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all gwdetectors
+          tags:
+            - gwdetectors
+          parameters:
+            - in: query
+              name: name
+              schema:
+                type: string
+              description: Filter by name (exact match)
```

Comment:
A partial match could be a real time saver in this case.

Reply:
good idea.